codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/Decoder.java

```diff
@@ -502,9 +505,6 @@ private void insertHeader(HeaderListener headerListener, byte[] name, byte[] val
 
     private void addHeader(HeaderListener headerListener, byte[] name, byte[] value,
                            boolean sensitive) {
-        if (name.length == 0) {
```

Comment:
Same reason, 0 is allowed for hpack, not for http.  (this is basically a technicality).